server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -1849,6 +1849,11 @@ public IndexCommit getIndexCommit() {
     }
 
     public void onSettingsChanged(TimeValue translogRetentionAge, ByteSizeValue translogRetentionSize, long softDeletesRetentionOps) {
+        onSettingsChanged(translogRetentionAge, translogRetentionSize, softDeletesRetentionOps, false);
+    }
+
+    public void onSettingsChanged(TimeValue translogRetentionAge, ByteSizeValue translogRetentionSize,
+                                  long softDeletesRetentionOps, boolean translogPruningByRetentionLease) {
```

Comment:
Unrelated to this PR; this noop is terrible design. We should make this method abstract and force derived classes to override w/ noop.

Reply:
Maybe we do that on main and backport to 1.x for fixing this in 1.2.

Reply:
I will add an issue to track this